syntax = "proto2";

package apollo.planning;

import "modules/planning/proto/st_boundary_config.proto";
// dp速度st的配置
message DpStSpeedConfig {                                                   // 在planning_config.pb.h文件中定义
  optional double total_path_length = 1 [ default = 0.1 ];                  // 总长度设计为149
  optional double total_time = 2 [ default = 3.0 ];                         // 总时间为7秒
  optional int32 matrix_dimension_s = 3 [ default = 100 ];                  // st矩阵的s维度为150
  optional int32 matrix_dimension_t = 4 [ default = 10 ];                   // t的维度为8

  optional double speed_weight = 5 [ default = 0.0 ];                       // 速度的权重为0
  optional double accel_weight = 6 [ default = 10.0 ];                      // 加速度的权重为10
  optional double jerk_weight = 7 [ default = 10.0 ];                       // 滚动向量的权重为10(即加速度的微分)
  optional double obstacle_weight = 8 [ default = 1.0 ];                    // 障碍物的权重为1.0
  optional double reference_weight = 9 [ default = 0.0 ];                   // 参考线的权重为0.0
  optional double go_down_buffer = 10 [ default = 5.0 ];                    // 下缓冲区的buffer为5.0
  optional double go_up_buffer = 11 [ default = 5.0 ];                      // 上缓冲去的buffer为5.0

  // obstacle cost config
  optional double default_obstacle_cost = 12 [ default = 1e10 ];            // 默认障碍物的代价为10的三次方

  // speed cost config
  optional double default_speed_cost = 13 [ default = 1.0 ];                // 默认速度的代价为10的三次方
  optional double exceed_speed_penalty = 14 [ default = 10.0 ];             // 超速的惩罚项为10
  optional double low_speed_penalty = 15 [ default = 2.5 ];                 // 低速的处罚为10
  optional double keep_clear_low_speed_penalty = 16 [ default = 10.0 ];     // keep clear可以不用管

  // accel cost config                                                      // 加速度的配置项
  optional double accel_penalty = 20 [ default = 2.0 ];                     // 加速度的惩罚项为1.0
  optional double decel_penalty = 21 [ default = 2.0 ];                     // 减速度的惩罚项为1.0

  // jerk cost config                                                       // 滚动惯量的配置项
  optional double positive_jerk_coeff = 30 [ default = 1.0 ];               // 正的滚动惯量为1.0
  optional double negative_jerk_coeff = 31 [ default = 300.0 ];             // 负的滚动惯量为1.0

  // other constraint                                                       // 其他约束项(最大加速度, 最大减速度)
  optional double max_acceleration = 40 [ default = 4.5 ];                  // 最大加速度为3.0
  optional double max_deceleration = 41 [ default = -4.5 ];                 // 最小减速度为4.0

  optional apollo.planning.StBoundaryConfig st_boundary_config = 50;        // st boundary的配置
}
